hazelcast-sql/src/test/java/com/hazelcast/sql/impl/expression/CastEndToEndTest.java

```diff
@@ -0,0 +1,369 @@
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package com.hazelcast.sql.impl.expression;
+
+import com.hazelcast.sql.impl.expression.math.ExpressionMath;
+import com.hazelcast.test.HazelcastParallelClassRunner;
+import com.hazelcast.test.annotation.ParallelJVMTest;
+import com.hazelcast.test.annotation.QuickTest;
+import org.junit.Test;
+import org.junit.experimental.categories.Category;
+import org.junit.runner.RunWith;
+
+import java.math.BigDecimal;
+
+import static com.hazelcast.sql.SqlColumnType.BIGINT;
+import static com.hazelcast.sql.SqlColumnType.BOOLEAN;
+import static com.hazelcast.sql.SqlColumnType.DECIMAL;
+import static com.hazelcast.sql.SqlColumnType.DOUBLE;
+import static com.hazelcast.sql.SqlColumnType.INTEGER;
+import static com.hazelcast.sql.SqlColumnType.OBJECT;
+import static com.hazelcast.sql.SqlColumnType.REAL;
+import static com.hazelcast.sql.SqlColumnType.SMALLINT;
+import static com.hazelcast.sql.SqlColumnType.TINYINT;
+import static com.hazelcast.sql.SqlColumnType.VARCHAR;
+
+@RunWith(HazelcastParallelClassRunner.class)
+@Category({QuickTest.class, ParallelJVMTest.class})
+public class CastEndToEndTest extends ExpressionEndToEndTestBase {
+
+    @Test
+    public void testBoolean() {
+        assertRow("cast(booleanTrue as boolean)", EXPR0, BOOLEAN, true);
+
+        assertParsingError("cast(byte1 as boolean)", "Cast function cannot convert value of type TINYINT to type BOOLEAN");
+        assertParsingError("cast(short1 as boolean)", "Cast function cannot convert value of type SMALLINT to type BOOLEAN");
+        assertParsingError("cast(int1 as boolean)", "Cast function cannot convert value of type INTEGER to type BOOLEAN");
+        assertParsingError("cast(long1 as boolean)", "Cast function cannot convert value of type BIGINT to type BOOLEAN");
+
+        assertParsingError("cast(float1 as boolean)", "Cast function cannot convert value of type REAL to type BOOLEAN");
+        assertParsingError("cast(double1 as boolean)", "Cast function cannot convert value of type DOUBLE to type BOOLEAN");
+
+        assertParsingError("cast(decimal1 as boolean)",
+                "Cast function cannot convert value of type DECIMAL(38, 38) to type BOOLEAN");
+        assertParsingError("cast(bigInteger1 as boolean)",
+                "Cast function cannot convert value of type DECIMAL(38, 38) to type BOOLEAN");
+
+        assertDataError("cast(string1 as boolean)", "VARCHAR value cannot be converted to BOOLEAN: 1");
+        assertRow("cast(stringFalse as boolean)", EXPR0, BOOLEAN, false);
+        assertDataError("cast(char1 as boolean)", "VARCHAR value cannot be converted to BOOLEAN: 1");
+
+        assertDataError("cast(object as boolean)", "Cannot convert OBJECT to BOOLEAN");
```

Comment:
What about adding this field to the `Record` in `ExpressionEndToEndTestBase`:
```java
public Object booleanTrueObject = Boolean.TRUE;
```
and then add the following line here:
```java
assertRow("cast(booleanTrueObject as boolean)", EXPR0, BOOLEAN, true);
```
Also for other types. It will check the conversion from OBJECT type.

Reply:
fixed in 9082f7d2b8af4d33049b70620c22f03e03150206